Fixes #3686 Adding Mandarin(traditional Chinese) translation to About US page

cychu42 opened the pull request:
## Issue This PR Addresses
Fixes #3686 

## Type of Change
- [ ] **Bugfix**: Change which fixes an issue
- [x] **New Feature**: Change which adds functionality
- [x] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description

This adds a Mandarin(traditional Chinese) translation to About US page(`overview.md`), and it also lists this translation to `docusaurus.config.js`.

## Steps to test the PR

Please navigate to `src/web/docusaurus/docs/overview.md` to find the `overview.md` document.

## Checklist

- [x] **Quality**: This PR builds and passes our npm test and works locally
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

RC-Lee commented on src/web/docusaurus/docs/overview.md:
```diff
@@ -4,131 +4,111 @@ sidebar_position: 0
+- 使用Traefik遷移單體式後臺到Microservices(90%完成)和API Gateway。
+- 改善Elasticsearch和Redis。
+- 從GatsbyJS到next.js全面轉移前臺。 
+- 重寫前臺的TypeScript。
```
```suggestion
- 以TypeScript重寫前端。
```

RC-Lee commented on src/web/docusaurus/docs/overview.md:
```diff
@@ -4,131 +4,111 @@ sidebar_position: 0
+- 改善Elasticsearch和Redis。
+- 從GatsbyJS到next.js全面轉移前臺。 
+- 重寫前臺的TypeScript。
+- 手動和自動(Dependabot)依賴更新和維護。
```
```suggestion
- 手動和自動(Dependabot)更新和維護依賴關係。
```

RC-Lee commented on src/web/docusaurus/docs/overview.md:
```diff
@@ -4,131 +4,111 @@ sidebar_position: 0
-We believe that one of the most rewarding parts of learning to work in the
-open source community is discovering that one can become part of the fabric
-of the web, find a voice, and build a following.
+Seneca的開源軟體開發其中一個關鍵的特色是重視分享我們正在做些甚麼、教育，以及藉由部落格發文來學習。
```
```suggestion
Seneca的開源開發計畫其中一個關鍵的特色是藉由部落格發文來重視分享我們正在做些甚麼、進行的教育，以及所受的學習。
```

cychu42 replied:
I'm not sure about 所受的學習, because I don't think you passively "receive" learn from other sources. You actively learn.

RC-Lee commented on src/web/docusaurus/docs/overview.md:
```diff
@@ -4,131 +4,111 @@ sidebar_position: 0
-open source community is discovering that one can become part of the fabric
-of the web, find a voice, and build a following.
+Seneca的開源軟體開發其中一個關鍵的特色是重視分享我們正在做些甚麼、教育，以及藉由部落格發文來學習。
+我們相信學習如何在開源軟體社群中作業最具獎勵的一部分是發現自己能夠成為網路的一份子、找到自己的聲音和建立一些追隨。
```
```suggestion
我們相信學習如何在開源社群中作業最具獎勵的一部分是發現自己能夠成為網路的一份子、找到自己的聲音和建立一些追隨。
```

RC-Lee commented on src/web/docusaurus/docs/overview.md:
```diff
@@ -4,131 +4,111 @@ sidebar_position: 0
-feed of blog posts from Seneca faculty and students working on open source in a
-single page. Our blog Planet currently lives at [Planet CDOT](https://telescope.cdot.systems/planet).
+為了能夠增加部落格的能見度，我們設立了一個開源部落格[Planet](<https://en.wikipedia.org/wiki/Planet_(software)>):
+一個將Seneca開源軟體開發的師生所發表的部落格文章化成單頁的集合體。
```
```suggestion
一個將開源開發的Seneca師生所發表的部落格文章化成單頁的集合體。
```

RC-Lee commented on src/web/docusaurus/docs/overview.md:
```diff
@@ -4,131 +4,111 @@ sidebar_position: 0
-various "feed" technologies (RSS, Atom, CDF) to allow blog posts from different
-platforms to be aggregated into a single page that was constantly updated with
-the latest posts by people within a particular community.
+在2000年代的早期，在像推特和臉書等社群軟體的崛起前，Planet解決了一個開源軟體社群的重要問題。
```
```suggestion
在2000年代的早期，在像推特和臉書等社群軟體的崛起前，Planet解決了一個開源社群的重要問題。
```

RC-Lee commented on src/web/docusaurus/docs/overview.md:
```diff
@@ -4,131 +4,111 @@ sidebar_position: 0
-involved in open source at Seneca, has become too difficult. Our current site
-often breaks, and needs manual interventions on a regular basis. Going forward,
-we need a new planet to call home.
+我們現在的Planet正在消亡中。我們用的軟體最後更新是13年前。當內含的程式碼正在往過去飄去，
```
```suggestion
我們現在的Planet正在消亡中。我們用的軟體最後更新是13年前。當內含的程式碼正在隨著時間流逝成為過去式，
```

RC-Lee commented on src/web/docusaurus/docs/overview.md:
```diff
@@ -4,131 +4,111 @@ sidebar_position: 0
-often breaks, and needs manual interventions on a regular basis. Going forward,
-we need a new planet to call home.
+我們現在的Planet正在消亡中。我們用的軟體最後更新是13年前。當內含的程式碼正在往過去飄去，
+我們的需求依然前進著。特別是因為在Seneca參予開源軟體開發的學生數量，維護既有的程式碼變得相當地困難。
```
```suggestion
我們的需求依然前進著。特別是因為在Seneca參予開源開發的學生數量，維護既有的程式碼變得相當地困難。
```

RC-Lee commented on src/web/docusaurus/docs/overview.md:
```diff
@@ -4,131 +4,111 @@ sidebar_position: 0
-- Staging (<https://dev.telescope.cdot.systems/>) and production (<https://telescope.cdot.systems/>) deployments
+[Telescope 1.0](https://github.com/Seneca-CDOT/telescope/releases/tag/1.0.0)實現了我們許多的初步目標，包含:
+
+- 一個單體式node.js後臺網路伺服器提供REST APIs和GraphQL。
```
```suggestion
- 一個提供REST APIs和GraphQL的單體式node.js後端網路伺服器。
```

RC-Lee commented on src/web/docusaurus/docs/overview.md:
```diff
@@ -4,131 +4,111 @@ sidebar_position: 0
+[Telescope 1.0](https://github.com/Seneca-CDOT/telescope/releases/tag/1.0.0)實現了我們許多的初步目標，包含:
+
+- 一個單體式node.js後臺網路伺服器提供REST APIs和GraphQL。
+- 一個用於摘要平行處理的node.js排隊服務。
```
```suggestion
- 一個用於摘要平行處理的node.js佇列服務。
```

RC-Lee commented on src/web/docusaurus/docs/overview.md:
```diff
@@ -4,131 +4,111 @@ sidebar_position: 0
+- 一個單體式node.js後臺網路伺服器提供REST APIs和GraphQL。
+- 一個用於摘要平行處理的node.js排隊服務。
+- 全面介面重整和設計。
+- 一個使用Material UI React部件的GatsbyJS前臺網路伺服器。
```
```suggestion
- 一個使用Material UI React部件的GatsbyJS前端網路伺服器。
```

RC-Lee commented on src/web/docusaurus/docs/overview.md:
```diff
@@ -4,131 +4,111 @@ sidebar_position: 0
+- 一個用於摘要平行處理的node.js排隊服務。
+- 全面介面重整和設計。
+- 一個使用Material UI React部件的GatsbyJS前臺網路伺服器。
+- 初步SAML2基礎的單一登入認證。　
```
```suggestion
- 以SSO認證初始化SAML2
```

cychu42 replied:
If it's 初始化, shouldn't the original english text say "Initialize"? Instead, the original says "Initial".

RC-Lee replied:
Makes sense. 
Its just the original sentence seems a little bit off to me.
What do you think of something like
```suggestion
- 開始以SAML2為基礎的SSO。　
```
If you think "初步" is better than "開始", then I guess keep it like that (maybe I haven't used Mandarin in too long :P).
I have a feeling "SSO" is better known than "單一登入認證", but if you want to keep it, maybe put it like "單一登入認證(SSO)"?

RC-Lee commented on src/web/docusaurus/docs/overview.md:
```diff
@@ -4,131 +4,111 @@ sidebar_position: 0
+- 一個使用Material UI React部件的GatsbyJS前臺網路伺服器。
+- 初步SAML2基礎的單一登入認證。　
+- Docker/Docker Compose基礎的容器管理。
+- 使用CircleCI和Travis CI的CI/CD管道。
```
```suggestion
- 使用CircleCI和Travis CI的CI/CD pipeline。
```

RC-Lee commented on src/web/docusaurus/docs/overview.md:
```diff
@@ -4,131 +4,111 @@ sidebar_position: 0
+- 初步SAML2基礎的單一登入認證。　
+- Docker/Docker Compose基礎的容器管理。
+- 使用CircleCI和Travis CI的CI/CD管道。
+- 使用Zeit Now的合併請求預覽。 
```
```suggestion
- 使用Zeit Now的Pull Request預覽。 
```

RC-Lee commented on src/web/docusaurus/docs/overview.md:
```diff
@@ -4,131 +4,111 @@ sidebar_position: 0
+- 一個Nginx反向代理和HTTP快取伺服器。
+- 用來管理使用Let’s Encrypt的SSL憑證的Certbot。
+- 一個node.js基礎的GitHub Webhook服務來自動管理基於
+- GitHub推事件和webhooks的部屬，以自動化整裝和生產軟體組建，包含藍綠部署。
```
```suggestion
- GitHub push event和webhooks的部屬，以自動化整裝和生產軟體組建，包含green/blue deployment。
```

RC-Lee commented on src/web/docusaurus/docs/overview.md:
```diff
@@ -4,131 +4,111 @@ sidebar_position: 0
+- 用來管理使用Let’s Encrypt的SSL憑證的Certbot。
+- 一個node.js基礎的GitHub Webhook服務來自動管理基於
+- GitHub推事件和webhooks的部屬，以自動化整裝和生產軟體組建，包含藍綠部署。
+- 整裝(<https://dev.telescope.cdot.systems/>)和生產(<https://telescope.cdot.systems/>)部署。
```
```suggestion
- Staging (<https://dev.telescope.cdot.systems/>)和Production(<https://telescope.cdot.systems/>)部署。
```

RC-Lee commented on src/web/docusaurus/docs/overview.md:
```diff
@@ -4,131 +4,111 @@ sidebar_position: 0
+- 改善SAML基礎認證、JWT授權和使用者註冊流程安全性。
+- 新的介面設計、Logo、CSS和Theming。
+- 改善無障礙和使用者體驗。
+- 使用Traefik遷移單體式後臺到Microservices(90%完成)和API Gateway。
```
```suggestion
- 使用Traefik遷移單體式後端到Microservices(90%完成)和API Gateway。
```

RC-Lee reviewed (changes requested):
Translating stuff is rough, so let's discuss and start with this. 
Basically
- For "open source" it might be fine to keep it just as "open source", but if we had to translate it, "開源" is fine in most cases, instead of "開源軟體"
- Front-end and Back-end should be "前端" and "後端"
- Some stuff I think are better left just as English 

Like I said, open for discussion, cause I'm not sure if we'll get a second reviewer.

RC-Lee commented on src/web/docusaurus/docs/overview.md:
```diff
@@ -4,131 +4,111 @@ sidebar_position: 0
-- [Telescope 3.0](https://blog.humphd.org/toward-telescope-3-0/) (in progress, April 2022)
+- [Telescope 1.0](https://blog.humphd.org/telescope-1-0-0-or-dave-is-once-again-asking-for-a-blog/) (2020年四月)
+- [Telescope 2.0](https://blog.humphd.org/telescope-2-0/) (2021年四月)
+- [Telescope 3.0](https://blog.humphd.org/toward-telescope-3-0/) (製作中，2020年四月)
```
```suggestion
- [Telescope 3.0](https://blog.humphd.org/toward-telescope-3-0/) (製作中，2022年四月)
```

cychu42 commented:
I initially felt the need to specify 軟體(software) in cases the term open source is used in other context, but I guess it’s pretty exclusive to software?
Another reason being that anyone who’s not familiar with open source or just programming in general in Mandarin can tell it’s about some kind of software.

As for "前端" and "後端" vs "前臺" and "後臺", I have seen both ways so I think it’s fine either way.

cychu42 commented:
Lastly, I'm not opposed to leaving some of those technical terms in English as you suggested, especially if you have experience with their convention in Mandarin. It’s not unheard of to just use the English terms for Mandarin speakers, when it comes to specialized terms.
I'm not entirely familiar with the convention for how some of the terms are translated, so I had to do a bit of digging around the internet to see how people refer to them in Mandarin. 合併請求 and 藍綠部署 show up a few times, so I’m inclined to say we can keep them, but rest of the terms you pointed out might be better left in English as you said.

RC-Lee commented:
> Lastly, I'm not opposed to leaving some of those technical terms in English as you suggested, especially if you have experience with their convention in Mandarin. It’s not unheard of to just use the English terms for Mandarin speakers, when it comes to specialized terms. I'm not entirely familiar with the convention for how some of the terms are translated, so I had to do a bit of digging around the internet to see how people refer to them in Mandarin. 合併請求 and 藍綠部署 show up a few times, so I’m inclined to say we can keep them, but rest of the terms you pointed out might be better left in English as you said.

I trust your research, so if you've seem them around a lot, then yes please do keep them as is. 
You seem to be missing a configuration on this line here in your PR. 
https://github.com/Seneca-CDOT/telescope/blob/76609baa6d2f52974ab3101eff0319890ec7532c/src/web/docusaurus/docusaurus.config.js#L44
Please also rebase when you're ready for review.

cychu42 commented:
Thank you for the review and suggestions. This is now ready for re-review.
I decided to keep 開源軟體 and implemented the changes we talked about.
I did change 合併請求 to 拉取請求, because the former actually means “merge request” (my bad!), while the later is how pull request is called.

RC-Lee commented:
The PR doesn't pass our CI, prettier specifically, please resolve them first.

cychu42 commented:
I managed to resovle the issue with Prettier.

RC-Lee commented:
Translations look fine. 
Pinging Docusaurus peeps for build related stuff 

cc: @sirinoks @cindyledev

cychu42 commented:
I'm looking at past translation PRs and the current project structure, I think that would be the general location.
However, I think you would want to put it under either a zh-cht folder for traditional Chinese or zh-tw folder for traditional Chinese (Taiwan). The zh-cn folder would be for simplified Chinese (China) instead.

sirinoks commented on src/web/docusaurus/docs/overview.md:
```diff
@@ -4,131 +4,111 @@ sidebar_position: 0
 import StarField from '@site/src/components/StarField';
 
-# Telescope Overview
+# Telescope 總覽
```
Looking at #3701 which worked out - yes. I think it has to be in the /i18n. The main one changed the original file. 

Just ran the build to check. You are correct. This is wrong. 
As you can see, the original English page changed to Chinese:
![image](https://user-images.githubusercontent.com/13803161/195602556-5936f21d-1c25-4e6b-9f0a-548b8a37c73d.png)

And the Chinese page is non existant:
![image](https://user-images.githubusercontent.com/13803161/195602322-45054bc5-f691-4353-8ca5-beb33d183f56.png)

sirinoks reviewed (changes requested):
Look at #3701 

Instead of changing the 
`src/web/docusaurus/docs/overview.md`

Change, or create yourself if you have to
`src/web/docusaurus/i18n/zh/docusaurus-plugin-content-docs/current/overview.md`

To test, also look at the "Steps to test PR". Just change uk to zh when npm start-ing

cychu42 commented:
OK. I reverted `src/web/docusaurus/docs/overview.md` to the original and moved my translation to `src/web/docusaurus/i18n/zh/docusaurus-plugin-content-docs/current/overview.md` instead, as requested.
Please let me know if you need anything else.

manekenpix reviewed (approved):
LGTM :+1: 

![image](https://user-images.githubusercontent.com/23108901/195665621-d921712f-d813-42de-9eb9-5c8e19cc356b.png)

RC-Lee reviewed (approved):
Built. Works.
![image](https://user-images.githubusercontent.com/32626950/195666740-250bde42-3d1c-42d5-becd-7fae59492cdb.png)
